dev/run-tests-jenkins.py

```diff
@@ -0,0 +1,253 @@
+
+
+def print_err(*args):
+    """
+    Given a set of arguments, will print them to the STDERR stream
+    """
+    print(*args, file=sys.stderr)
+
+
+def post_message(mssg, comments_url):
+    http_code_header = "HTTP Response Code: "
+    posted_message = json.dumps({"body": mssg})
+
+    print("Attempting to post to Github...")
+
+    # we don't want to call `run_cmd` here as, in the event of an error, we DO NOT
+    # want to print the GITHUB_OAUTH_KEY into the public Jenkins logs
+    curl_proc = subprocess.Popen(['curl',
+                                  '--silent',
+                                  '--user', 'x-oauth-basic:' + os.environ['GITHUB_OATH_KEY'],
+                                  '--request', 'POST',
+                                  '--data', posted_message,
+                                  '--write-out', http_code_header + '%{http_code}',
+                                  '--header', 'Content-Type: application/json',
+                                  comments_url],
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE)
+    curl_stdout, curl_stderr = curl_proc.communicate()
+    curl_returncode = curl_proc.returncode
+    # find all lines relevant to the Github API response
+    api_response = "\n".join([l for l in curl_stdout.split('\n')
+                              if l and not l.startswith(http_code_header)])
+    # find the line where `http_code_header` exists, split on ':' to get the
+    # HTTP response code, and cast to an int
+    http_code = int(curl_stdout[curl_stdout.find(http_code_header):].split(':')[1])
+
+    if not curl_returncode == 0:
+        print_err("Failed to post message to GitHub.")
+        print_err(" > curl_status:", curl_returncode)
+        print_err(" > curl_output:", curl_stdout)
+        print_err(" > data:", posted_message)
+
+    if http_code and not http_code == 201:
+        print_err(" > http_code:", http_code)
+        print_err(" > api_response:", api_response)
+        print_err(" > data:", posted_message)
+
+    if curl_returncode == 0 and http_code == 201:
+        print(" > Post successful.")
+
+
+def send_archived_logs():
+    print("Archiving unit tests logs...")
+
+    log_files = run_cmd(['find', '.',
+                         '-name', 'unit-tests.log',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.failed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.hiveFailed',
+                         '-o', '-path', './sql/hive/target/HiveCompatibilitySuite.wrong'],
+                        return_output=True)
+
+    if log_files:
+        log_archive = "unit-tests-logs.tar.gz"
+
+        run_cmd(['tar', 'czf', log-archive] + log_files.strip().split('\n'))
+
+        jenkins_build_dir = os.environ["JENKINS_HOME"] + "/jobs/" + os.environ["JOB_NAME"] +
+        "/builds/" + os.environ["BUILD_NUMBER"]
```

Comment:
I think that `pep8` would complain about this continuation line being underindented.